packages/typespec-client-generator-core/test/types.test.ts

```diff
@@ -956,7 +959,13 @@ describe("typespec-client-generator-core: types", () => {
       strictEqual(clientProjectedProp.kind, "property");
       strictEqual(clientProjectedProp.serializedName, "clientProjectedName");
 
-      // wire name test
+      // wire name test with encoded and projected
+      const jsonEncodedProp = sdkModel.properties.find(
+        (x) => x.kind === "property" && x.serializedName === "encodedWireName"
+      )!;
```

Comment:
find could return undefined and cause this to crash. It may be worth removing the `!` and handling undefined and throw an error that can help when debugging